### PR TITLE
[i18n] CTB Add mutation API

### PR DESCRIPTION
**Pull request by soupette**
Signed-off-by: soupette <cyril.lpz@gmail.com>

### What does it do?

It allows a plugin to change the shape a content type before submission.

### Why is it needed?

In order a for plugin to interact better with the content type builder.

### CODE REVIEW

File: packages/strapi-plugin-content-type-builder/admin/src/utils/formAPI.js [hunk 1]
```diff
@@ -92,6 +96,17 @@ const formsAPI = {
 
     return initShape.shape({ pluginOptions: yup.object().shape(pluginOptionsShape) });
   },
+  sanitizeContentTypeSchema(data, initialData) {
+    let sanitizedData = cloneDeep(data);
+
+    const refData = cloneDeep(initialData);
+
+    this.contentTypeSchemaSanitizers.forEach(cb => {
+      sanitizedData = cb(sanitizedData, refData);
```
Comment (mfrachet) [inline packages/strapi-plugin-content-type-builder/admin/src/utils/formAPI.js:105]: You're re-affecting sanitizedData on each iteration, is this a wanted behaviour?

File: packages/strapi-plugin-i18n/admin/src/utils/sanitizeCTBContentTypeSchema.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import { has, get, omit } from 'lodash';
+import LOCALIZED_FIELDS from './localizedFields';
+
+const localizedPath = ['pluginOptions', 'i18n', 'localized'];
+
+const sanitizeCTBContentTypeSchema = (nextSchema, prevSchema) => {
```
Comment (mfrachet) [inline packages/strapi-plugin-i18n/admin/src/utils/sanitizeCTBContentTypeSchema.js:6]: I'm not sure sanitizing is a valid term in this situation 🤔 . For me, sanitizing is about transforming invalid data and make sure they are valid.

For instance, when a user is able to fill some data, that data can't be trusted (for security reasons for example), we can eventually say that this data can be "dirty". Sanitizing, for me, is the process of making sure this potentially dirty data can't affect our system.

I'm maybe missing something, but in this scenario, I'm feeling this is more a transformation on the schema to make it have the shape we want it to have no?

**Review by mfrachet — COMMENTED** [review]
This looks good to me 😊 . I've added comments about eventually naming things, but otherwise, looks good :D

File: packages/strapi-plugin-content-type-builder/admin/src/utils/formAPI.js (hunk 1 above)
Comment (soupette) [reply packages/strapi-plugin-content-type-builder/admin/src/utils/formAPI.js:105]: Yes, we need to let the ability for a plugin to mutate the schema before the form submission.

File: packages/strapi-plugin-i18n/admin/src/utils/sanitizeCTBContentTypeSchema.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-plugin-i18n/admin/src/utils/sanitizeCTBContentTypeSchema.js:6]: I agree the naming is incorrect, I was struggling to find a good naming, what about `mutateCTBContentTypeSchema`?

Comment (mfrachet) [reply packages/strapi-plugin-i18n/admin/src/utils/sanitizeCTBContentTypeSchema.js]: Yes, this sounds better to me. Also, maybe defining what is the "type" of the input and the "type" of the output would help finding a great name; WYT? 😊

**Pull request merged**